src/node_api_embedding.h

```diff
@@ -0,0 +1,234 @@
+#ifndef SRC_NODE_API_EMBEDDING_H_
+#define SRC_NODE_API_EMBEDDING_H_
+
+#include "node_api.h"
+
+EXTERN_C_START
+
+typedef struct node_api_env_options__* node_api_env_options;
+
+typedef enum {
+  node_api_platform_no_flags = 0,
```

Comment:
since a bunch of them seem to disable specific flags should there be an all_flags, or are they all on by default and then there are no/disable flags only?

Reply:
I have changed the approach since our last Node-API meeting. These flags are 1-to-1 mapping to the flags that Node.js has. The default is the `no_flags` configuration. Then, embedders can disable some default Node.js features.
I can add an alias for the `no_flags` as a `default_flags`.